src/Azure/Orleans.Persistence.AzureStorage/Providers/Storage/AzureBlobStorageOptions.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+using System.Collections.Generic;
+
+namespace Orleans.Hosting
+{
+    public class AzureBlobStorageOptions
+    {
+        /// <summary>
+        /// Azure data connection string
+        /// </summary>
+        public string DataConnectionString { get; set; }
```

Comment:
add OptionsValidator to validate options. For example, Connectionstring is required for AzureBlobStorageOptions. 

And also we need to be consistent with the naming. It was called ConnectionString in Membership and gateway providers. Since `Data` is redundant here.

Reply:
I agree that we should be consistent. Here I was consistent with what was done in `AzureTableStorageOptions`. I believe that the name "DataConnectionString" was used as a parameter name in the azure sdk before, I assume that's why this name was used.

I don't care what name we use in the end but yes we should be consistent.

Reply:
I updated  that name to be ConnectionString in #4032 . So we are consistent ;)